internal/install/_static/Dockerfile.terraform_deployer

```diff
@@ -2,15 +2,15 @@ FROM hashicorp/terraform:light as terraform
 
 FROM python:3-alpine
 
-# required by gcloud SDK
 RUN apk add --no-cache git openssh curl
 
+# Announcements for new releases: https://groups.google.com/g/google-cloud-sdk-announce
 ENV GCLOUD_SDK_VERSION 369.0.0
 ENV PATH $PATH:/usr/local/gcloud/google-cloud-sdk/bin
-RUN curl "https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cloud-sdk-$GCLOUD_SDK_VERSION-linux-x86_64.tar.gz" > /tmp/google-cloud-sdk.tar.gz \
+RUN curl --silent "https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cloud-sdk-$GCLOUD_SDK_VERSION-linux-x86_64.tar.gz" > /tmp/google-cloud-sdk.tar.gz \
   && mkdir -p /usr/local/gcloud \
-  && tar -C /usr/local/gcloud -xvf /tmp/google-cloud-sdk.tar.gz \
-  && /usr/local/gcloud/google-cloud-sdk/install.sh -q --override-components="bq" \
+  && tar -C /usr/local/gcloud -xf /tmp/google-cloud-sdk.tar.gz \
+  && /usr/local/gcloud/google-cloud-sdk/install.sh -q --override-components="bq" &>/dev/null \
```

Comment:
According to the public [docs](https://cloud.google.com/sdk/docs/downloads-interactive#silent), there is `--disable-prompts` flag. We can give it a try and let the installer decide what's rubbish and what isn't :)

Reply:
updated. Where can I look at the logs? I've not been able to find the pipeline where this log is displayed :(

Reply:
Do you mean [this](https://beats-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Felastic-package/detail/PR-665/2/pipeline)?

```
[2022-01-26T14:00:31.542Z]  ---> Running in f3dbb078bd82
[2022-01-26T14:00:41.632Z] Welcome to the Google Cloud SDK!
[2022-01-26T14:00:41.632Z] WARNING: You appear to be running this script as root. This may cause 
[2022-01-26T14:00:41.632Z] the installation to be inaccessible to users other than the root user.
[2022-01-26T14:00:43.561Z] usage: install.py [-h] [--usage-reporting USAGE_REPORTING]
[2022-01-26T14:00:43.561Z]                   [--screen-reader SCREEN_READER] [--rc-path RC_PATH]
[2022-01-26T14:00:43.561Z]                   [--command-completion COMMAND_COMPLETION]
[2022-01-26T14:00:43.561Z]                   [--path-update PATH_UPDATE] [--disable-installation-options]
[2022-01-26T14:00:43.561Z]                   [--override-components [OVERRIDE_COMPONENTS [OVERRIDE_COMPONENTS ...]]]
[2022-01-26T14:00:43.561Z]                   [--additional-components ADDITIONAL_COMPONENTS [ADDITIONAL_COMPONENTS ...]]
[2022-01-26T14:00:43.561Z]                   [--quiet] [--install-python INSTALL_PYTHON]
[2022-01-26T14:00:43.561Z] install.py: error: unrecognized arguments: --disable-prompts
[2022-01-26T14:00:47.792Z] Service 'terraform' failed to build: The command '/bin/sh -c curl --silent "https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cloud-sdk-$GCLOUD_SDK_VERSION-linux-x86_64.tar.gz" > /tmp/google-cloud-sdk.tar.gz   && mkdir -p /usr/local/gcloud   && tar -C /usr/local/gcloud -xf /tmp/google-cloud-sdk.tar.gz   && /usr/local/gcloud/google-cloud-sdk/install.sh -q --override-components="bq" --disable-prompts   && rm /tmp/google-cloud-sdk.tar.gz' returned a non-zero code: 2
```

Reply:
It looks there are different options available, sorry for linking wrong docs. Could you please adjust the command?

Reply:
`--disable-prompts` is not available with the installation method in use, the tests for https://github.com/elastic/elastic-package/pull/665/commits/8e505f2353f21fa23ef6ce0e4e989675540ce553 broke with:

```
[2022-01-26T14:00:41.632Z] Welcome to the Google Cloud SDK!
[2022-01-26T14:00:41.632Z] WARNING: You appear to be running this script as root. This may cause 
[2022-01-26T14:00:41.632Z] the installation to be inaccessible to users other than the root user.
[2022-01-26T14:00:43.561Z] usage: install.py [-h] [--usage-reporting USAGE_REPORTING]
[2022-01-26T14:00:43.561Z]                   [--screen-reader SCREEN_READER] [--rc-path RC_PATH]
[2022-01-26T14:00:43.561Z]                   [--command-completion COMMAND_COMPLETION]
[2022-01-26T14:00:43.561Z]                   [--path-update PATH_UPDATE] [--disable-installation-options]
[2022-01-26T14:00:43.561Z]                   [--override-components [OVERRIDE_COMPONENTS [OVERRIDE_COMPONENTS ...]]]
[2022-01-26T14:00:43.561Z]                   [--additional-components ADDITIONAL_COMPONENTS [ADDITIONAL_COMPONENTS ...]]
[2022-01-26T14:00:43.561Z]                   [--quiet] [--install-python INSTALL_PYTHON]
[2022-01-26T14:00:43.561Z] install.py: error: unrecognized arguments: --disable-prompts
```

Reverting the commit

Reply:
Look at the usage. It seems that you can try using other flags: `--quiet`, `--disable-installation-options`.

Reply:
`--quiet` was already used. `--disable-installation-options` does not change the output. 

I've not been able to find a better alternative than redirecting to `/dev/null`. Also part of the installation output is sent to `stderr` (the colored lines), so the only way to suppress output is to redirect both `stdout` and `stderr` to `/dev/null`.

Reply:
I see, thanks for checking this. In this case, I would be for not hiding any possible errors by flushing them to `/dev/null`. Let's remove output redirection, but keep the "quiet mode"

Reply:
@endorama I looked at [docs](https://cloud.google.com/sdk/docs/install#other_installation_options) once again and it recommends  not executing `install.sh` for CI setups:

> Using Cloud SDK with scripts or Continuous Integration/Deployment? Download a versioned archive for a non-interactive installation of a specific version of Cloud SDK.

Step 4 of the [instruction](https://cloud.google.com/sdk/docs/downloads-versioned-archives#installation_instructions) says that `install.sh` just sets the PATH accordingly and installs bash-completion, which means we really don't need to do this here.

Could you please try again without calling `install.sh` at all? I suppose you need to unpack the archive in the correct place instead of `/tmp`.

PS We will definitely need a package system test to verify if it works as expected.

Reply:
Sorry, arriving late to the party. Now that this image is having dependencies with more complex installation methods, wdyt about changing the base image to an ubuntu one and [use apt](https://cloud.google.com/sdk/docs/install#deb) (with the `--quiet` flag :)) to install it?

Reply:
> wdyt about changing the base image to an ubuntu one and use apt

I don't have a preference here as long as we keep it working. Personally, I prefer smaller images (like alpine), Ubuntu has always seemed full-blown to me.

Reply:
> I prefer smaller images (like alpine), Ubuntu has always seemed full-blown to me.

Regarding size, I think that the base image is the lesser of the problems once we include the gcloud sdk :slightly_smiling_face: 
```
eb91706519b5   5 minutes ago   /bin/sh -c curl "https://dl.google.com/dl/cl…   1.05GB    
```
Apart of that, the ubuntu base image is quite small (72MB, ~100MB with python 3), and you probably already have it because is the base image of current versions of the stack.

Reply:
Looks more readable, and seems to take less space (tough maybe more components need to be installed with the optional packages).
```
FROM ubuntu:20.04

RUN apt-get -q update && apt-get install -yq curl apt-transport-https ca-certificates gnupg && apt-get clean

ADD google-cloud-sdk.list /etc/apt/sources.list.d/google-cloud-sdk.list
RUN curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add -

ENV GCLOUD_SDK_VERSION 369.0.0-0

RUN apt-get -q update && \
    apt-get -yq install google-cloud-sdk=${GCLOUD_SDK_VERSION} && \
    apt-get clean
```
```
da5b98a5dec3   About a minute ago   /bin/sh -c apt-get -q update &&     apt-get …   620MB     
```

Reply:
```
eb91706519b5   5 minutes ago   /bin/sh -c curl "https://dl.google.com/dl/cl…   1.05GB    
```

Yeah, that's something I wanted originally to omit, but it looks like it may not be so easy unless we modify the service deployer image to pull it in runtime?

Reply:
@mtojek I'm already using the versioned archive installation. `install.sh` triggers a python script that takes care of installing other components so I don't think can be safely skipped.

> Could you please try again without calling install.sh at all? I suppose you need to unpack the archive in the correct place instead of /tmp.

I tried it but the result is not good. This is what happens when running `gcloud` in the docker container without running install:

```
/workspace # gcloud
ERROR: (gcloud) Command name argument expected.
ERROR: gcloud failed to load (gcloud.interactive): Problem loading gcloud.interactive: cannot import name 'Mapping' from 'collections' (/usr/local/lib/python3.10/collections/__init__.py).

This usually indicates corruption in your gcloud installation or problems with your Python interpreter.

[...]

> PS We will definitely need a package system test to verify if it works as expected.

This is in the works!

I'm moving forward to using Ubuntu as base image.

Reply:
I updated the image to use Ubuntu.

These are the dimensions for the two images:
```
ubuntu        978MB
alpine        594MB
```

@jsoriano the gcloud SDK when installed from a versioned archive is a lot less that 1.05GB, as is possible to specify which components to install thus reducing the overall size. 

I'm happy with both, let me know what do you think of the current change :)

Reply:
Much better now and managed in a normal way :)

I will drop few nit-picks.